packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -510,6 +510,16 @@ export default function useAutocomplete(props) {
       return;
     }
 
+    // Do not sync highlighted index when user don't want to highlight for freeSolo
+    if (!autoHighlight && freeSolo) {
+      // Reset highlighted index if user navigated using keyboard in previous input
+      if (highlightedIndexRef.current !== -1) {
+        setHighlightedIndex({ index: -1 });
```

Comment:
```suggestion
        changeHighlightedIndex({ diff: 'reset' });
```
so that `onHighlightChange` is called and provides the correct parameters.